Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/msg/IConceptMessage.java

```diff
@@ -0,0 +1,65 @@
+package edu.asu.conceptpower.rest.msg;
+
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.validation.ObjectError;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import edu.asu.conceptpower.core.ConceptEntry;
+import edu.asu.conceptpower.core.ConceptType;
+
+/**
+ * This class acts as an interface for all concept message. Currently concepts
+ * are returned in json and xml format. So we have XmlConceptMessage and
+ * JsonConceptMessage implementing this class
+ * 
+ * @author karthik
+ *
+ */
+public interface IConceptMessage {
+
+    /**
+     * This method checks for the validation errors and calls
+     * getAllConceptEntriesAndPaginationDetails for processing.
+     * 
+     * @param entries
+     * @return
+     * @throws JsonProcessingException
+     */
+    public String getAllConceptEntries(Map<ConceptEntry, ConceptType> entries) throws JsonProcessingException;
+
+    /**
+     * This method fetches the concept entries with the specified pagination. If
+     * pagination is not supplied all the entries are fetched.
```

Comment:
"all the entries should be fetch"

again, since this is an interface you can only say what should be done not what is done.